src/internal/connector/graph/concurrency_limiter_test.go

```diff
@@ -0,0 +1,70 @@
+package graph
+
+import (
+	"math/rand"
+	"net/http"
+	"net/http/httptest"
+	"sync"
+	"testing"
+	"time"
+
+	khttp "github.com/microsoft/kiota-http-go"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/alcionai/corso/src/internal/tester"
+)
+
+type ConcurrencyLimiterTestSuite struct {
+	tester.Suite
+}
+
+func TestConcurrencyLimiterSuite(t *testing.T) {
+	suite.Run(t, &ConcurrencyLimiterTestSuite{Suite: tester.NewUnitSuite(t)})
+}
+
+func (suite *ConcurrencyLimiterTestSuite) TestConcurrencyLimiterMiddleware() {
+	t := suite.T()
+	maxConcurrentRequests := 4
+
+	InitializeConcurrencyLimiter(maxConcurrentRequests)
+
+	cl := GetConcurrencyLimiter()
+	client := khttp.GetDefaultClient(cl)
+
+	sem := make(chan struct{}, maxConcurrentRequests)
+	reqHandler := func(w http.ResponseWriter, r *http.Request) {
+		select {
+		case sem <- struct{}{}:
```

Comment:
Wow, it's been a while.  I actually forgot about non-blocking writes and thought this was a bug at first (: